datasource/opennms-direct/src/main/java/org/opennms/alec/datasource/opennms/jvm/DirectAlarmDatasource.java

```diff
@@ -165,7 +163,7 @@ public void handleNewOrUpdatedAlarm(org.opennms.integration.api.v1.model.Alarm a
                 }
             }
         } else {
-            final Situation oceSituation = mapper.toSituation(existingAlarm);
+            final Situation oceSituation = mapper.toSituation(alarm);
             return (a) ->  situationHandlers.forEach(h -> h.onSituation(oceSituation));
         }
         return (a) -> {};
```

Comment:
Just pointing out all of these `Consumer<Void>` should be `Runnable`s instead and the methods that return them should return `Optional<Runnable>` so we don't have to pass empty lambda's around.

Sending explicit null to a void consumer doesn't make much sense and adding noop lambdas to the lists of callbacks doesn't really make sense either.

Leave it up to you if you want to clean those up or not given its outside the scope of this specific bug fix.

Reply:
Since I'm not familiar with Alec code, I would leave it for you guys to refactor.